CIP-DappConnector/CIP-dapp-connector.md

```diff
@@ -237,6 +243,27 @@ Errors: `APIError`, `TxSendError`
 
 As wallets should already have this ability, we allow dApps to request that a transaction be sent through it. If the wallet accepts the transaction and tries to send it, it shall return the transaction id for the dApp to track. The wallet is free to return the `TxSendError` with code `Refused` if they do not wish to send it, or `Failure` if there was an error in sending it (e.g. preliminary checks failed on signatures).
 
+## Events
+
+In addition to the API methods that are actively called, the connector also must emit the following events. All methods events are required to be implemented.
+
+TODO: event emission method? Possible methods:
```

Comment:
How should we do this? I'm not usually a js developer so I'm not sure which is best.

Reply:
I would actually add the events to the api, not to the wallet. You don't call the events before you have the api object.
I would emit the event in a callback, so either `api.onAccountChange((addresses) => {...})` or `api.on("accountChange",(addresses) => {...})`.Latter how it's used in the ethereum API.